lib/livebook/teams/app_deployment.ex

```diff
@@ -0,0 +1,100 @@
+defmodule Livebook.Teams.AppDeployment do
+  use Ecto.Schema
+  alias Livebook.FileSystem
+
+  @file_extension ".zip"
+
+  @type t :: %__MODULE__{
+          id: String.t() | nil,
+          filename: String.t() | nil,
+          slug: String.t() | nil,
+          sha: String.t() | nil,
+          title: String.t() | nil,
+          deployment_group_id: String.t() | nil,
+          file: FileSystem.File.t() | String.t() | nil,
+          deployed_by: String.t() | nil,
+          deployed_at: NaiveDateTime.t() | nil
+        }
+
+  @primary_key {:id, :string, autogenerate: false}
+  embedded_schema do
+    field :filename, :string
+    field :slug, :string
+    field :sha, :string
+    field :title, :string
+    field :deployment_group_id, :string
+    field :file, :map
+    field :deployed_by, :string
+
+    timestamps(updated_at: nil, inserted_at: :deployed_at)
+  end
+
+  @doc """
+  Creates a new app deployment from session with compressed file.
+  """
+  @spec new(Livebook.Session.Data.t()) :: {:ok, t()} | {:error, FileSystem.error()}
+  def new(%Livebook.Session.Data{} = data) do
+    attachments =
+      data.notebook.file_entries
+      |> Enum.filter(&(&1.type == :file))
+      |> Enum.map(& &1)
+
+    with {:ok, file_system} <- FileSystem.File.fetch_file_system(data.file),
+         {:ok, file} <- zip_files(file_system, data.file, attachments),
+         {:ok, content} <- FileSystem.File.read(file) do
+      md5_hash = :crypto.hash(:md5, content)
+      shasum = Base.encode16(md5_hash, case: :lower)
```

Comment:
I wonder if the sha should of the encrypted content? Also, I would probably encrypt it here. This means we don't need to store the `file`?

Reply:
Since we're encrypting before sending to Teams inside the `Requests` module, I would keep it there just for standardization where we encrypt data.